sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyManager.java

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import javax.net.ssl.X509ExtendedKeyManager;
+import java.net.Socket;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Principal;
+import java.security.PrivateKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.INFO;
+import static java.util.logging.Level.WARNING;
+
+/**
+ * The KeyVault variant of the X509ExtendedKeyManager.
+ */
+public class KeyVaultKeyManager extends X509ExtendedKeyManager {
+
+    /**
+     * Stores the logger.
+     */
+    private static final Logger LOGGER = Logger.getLogger(KeyVaultKeyManager.class.getName());
+
+    /**
+     * Stores the keystore.
+     */
+    private final KeyStore keystore;
+
+    /**
+     * Stores the password.
+     */
+    private final char[] password;
+
+    /**
+     * Constructor.
+     *
+     * @param keystore the keystore.
+     * @param password the password.
+     */
+    public KeyVaultKeyManager(KeyStore keystore, char[] password) {
+        LOGGER.entering("KeyVaultKeyManager", "<init>", new Object[] { keystore, password });
+        this.keystore = keystore;
+        this.password = password;
+    }
+
+    @Override
+    public String chooseClientAlias(String[] keyType, Principal[] issuers, Socket socket) {
+        LOGGER.entering(
+            "KeyVaultKeyManager",
+            "chooseClientAlias",
+            new Object[] { keyType, issuers, socket }
+        );
+        String alias = null;
+        try {
+            /*
+             * If we only have one alias and the keystore type is not 'AzureKeyVault'
+             * return that alias as a match.
+             */
+            if (!keystore.getProvider().getName().equals("AzureKeyVault")
+                && keystore.size() == 1) {
+                alias = keystore.aliases().nextElement();
+            }
+        } catch (KeyStoreException kse) {
+            LOGGER.log(WARNING, "Unable to choose client alias", kse);
```

Comment:
No as that would cause the application potentially to be very chatty. If the customer would need to know if it is indeed returning `null` they can change the logging level.